### PR TITLE
Removing TableTools, fixing default stream mount

### PR DESCRIPTION
**Pull request by zklosko**
In this PR, I've removed the TableTools Flash export buttons and functionality, fixed the default logo display, and fixed the default stream mountpoint. At the moment, there is no easy drop-in fix for the Playout History export buttons, but the possibility to create a custom exporter using Vanilla JavaScript and plugin libraries remains. Updating DataTables to 1.10 from 1.9.x completely breaks the interface.

~This PR also removes the podcast player from the microsite (https://github.com/LibreTime/libretime/issues/1121#issuecomment-751420795).~

~Some PHP rewriting was done as well, removing unnecessary echo functions.~

Issues: #1133 #518

### Screenshots

Screen Shot 2020-12-26 at 5 33 57 PM
Screen Shot 2020-12-26 at 5 37 05 PM
Screen Shot 2020-12-26 at 5 36 42 PM

### CODE REVIEW

File: airtime_mvc/application/views/scripts/index/index.phtml [hunk 1]
```diff
@@ -306,21 +270,12 @@
...
             $("#player_iframe").contents().find('.bottom_bar').append("<div class='about_us_btn button' data-tab='tab-2'></div>");
-
-            <?php
-            // Is there a better way to do this?
-            if ($this->displayRssTab) {?>
-                $("#player_iframe").contents().find('.bottom_bar').append("<div class='station_rss_btn button' data-tab='tab-4'></div>");
-            <?php
-            }
-            ?>
+            $("#player_iframe").contents().find('.bottom_bar').append("<div class='station_rss_btn button' data-tab='tab-4'></div>");
```
Comment (hairmare) [inline airtime_mvc/application/views/scripts/index/index.phtml:278]: `displayRssTab` is still getting set in `IndexController`:
https://github.com/LibreTime/libretime/blob/6c4b1b5e71f8de260a1b04ee3f04c7489b1832bd/airtime_mvc/application/controllers/IndexController.php#L83
It is managed in `PreferenceController`:
https://github.com/LibreTime/libretime/blob/6c4b1b5e71f8de260a1b04ee3f04c7489b1832bd/airtime_mvc/application/controllers/PreferenceController.php#L111
It looks like this change will remove the ability to hide the rss "tab" or am I missing something any you found "a better way to do this"?

File: airtime_mvc/application/views/scripts/index/index.phtml [hunk 2]
```diff
@@ -43,261 +23,245 @@
...
-        <div id="tab-4" class="podcast tab_content">
-            <div class="rss-title-wrapper">
-                <div id="rss-subscribe">
-                    <a target="_blank" href="<?php echo $this->stationPodcastRssUrl ?>" class="rss_icon"><span><?php echo _("Subscribe") ?></span><img src="css/radio-page/img/rss.png"></a>
-                </div>
-                <div id="rss-title">
-                    <?php echo $this->podcastTitle ?>
-                </div>
+    <div id="tab-4" class="podcast tab_content">
+        <div class="rss-title-wrapper">
```
Comment (hairmare) [inline airtime_mvc/application/views/scripts/index/index.phtml:27]: Not checking `displayRssTab ` seems to break when stations do not have a public podcast (see above for details).

File: airtime_mvc/application/views/scripts/login/index.phtml [hunk 3]
```diff
@@ -1,12 +1,7 @@
 <div class="login_box">
-    <div class="logobox">&nbsp;</div>
-        <h2><?php echo _("Login") ?></h2>
+    <div class="logobox"></div>
+        <h2>Login</h2>
     <div id="login" class="login-content clearfix">
-        <?php if(isset($this->demo) && $this->demo == 1){?>
```
Comment (hairmare) [inline airtime_mvc/application/views/scripts/login/index.phtml:5]: It feels right to remove the demo code. We should remove the correponding code in `LoginController` as well.
https://github.com/LibreTime/libretime/blob/6c4b1b5e71f8de260a1b04ee3f04c7489b1832bd/airtime_mvc/application/controllers/LoginController.php#L100-L102

**Review by hairmare — CHANGES_REQUESTED** [review]
We can re-add export possibilities once we switch to the vue frontend (what I'm preparing for playouthistory in #1122).

Other than some small i18n issues and a few bits of dead code this is looking good so far.

File: airtime_mvc/application/views/scripts/index/index.phtml [hunk 4]
```diff
@@ -1,40 +1,20 @@
...
-                <span><?php echo _('Login'); ?></span>
-                <span class='login-img'></span>
-            </a>
-        </div>
-    <?php endif; ?>
+    ?>
+
+    <div class='login-btn'>
+        <a href='/login' target='_parent'>
+            <span>Login</span>
```
[the remark this answers is not in the document]
Comment (zklosko) [reply airtime_mvc/application/views/scripts/index/index.phtml:11]: I completely agree, but how is this being translated? I didn't see a i8 something call attached to this one.

File: airtime_mvc/application/views/scripts/index/index.phtml (hunk 1 above)
Comment (zklosko) [reply airtime_mvc/application/views/scripts/index/index.phtml:278]: Thanks for mentioning it was in the controllers. There doesn't appear to be a setting for it anywhere in the interface, so I removed the logic for it.

File: airtime_mvc/application/views/scripts/index/index.phtml (hunk 4 above)
[the remark this answers is not in the document]
Comment (hairmare) [reply airtime_mvc/application/views/scripts/index/index.phtml:11]: I think it gets translated to the os default language which is english in most cases (the lang switcher on the ui happens after this gets render afaict).

File: airtime_mvc/application/views/scripts/index/index.phtml (hunk 1 above)
Comment (hairmare) [reply airtime_mvc/application/views/scripts/index/index.phtml:278]: it's this thing on `/podcast/station`:

image

File: airtime_mvc/application/views/scripts/index/index.phtml (hunk 4 above)
[the remark this answers is not in the document]
Comment (zklosko) [reply airtime_mvc/application/views/scripts/index/index.phtml:11]: So the php echo call is still needed in all cases?

[the remark this answers is not in the document]
Comment (hairmare) [reply airtime_mvc/application/views/scripts/index/index.phtml:11]: Yeah it calls the `_()` function which does the actual translating. Using the underscore as a function name for this is a de-facto standard when working with gettext.

[the remark this answers is not in the document]
Comment (zklosko) [reply airtime_mvc/application/views/scripts/index/index.phtml:11]: Gotcha. Thanks for filling me in! I'll work on putting it back now.

File: airtime_mvc/application/views/scripts/index/index.phtml (hunk 2 above)
Comment (zklosko) [reply airtime_mvc/application/views/scripts/index/index.phtml:27]: I meant to only remove the extra login form 🤦

File: airtime_mvc/application/views/scripts/index/index.phtml (hunk 1 above)
Comment (zklosko) [reply airtime_mvc/application/views/scripts/index/index.phtml:278]: Gotcha

File: airtime_mvc/application/views/scripts/embed/player.phtml [hunk 5]
```diff
@@ -109,7 +109,7 @@
 
         Html5Player.prototype.play = function() {
             console.log('play');
-            playerhtml5_audio.src = this.settings.url+'?'+Math.floor(Math.random() * Math.floor(100000));
+            playerhtml5_audio.src = this.settings.url; //+'?'+Math.floor(Math.random() * Math.floor(100000));
```
Comment (hairmare) [inline airtime_mvc/application/views/scripts/embed/player.phtml:112]: I think this was intended as a cache buster but it looks broken anyway.

**Review by hairmare — DISMISSED** [review]

**Comment by paddatrapper** [comment]
I can't test this thanks to #1142. @hairmare did you have a chance to test this?

**Comment by hairmare** [comment]
> I can't test this thanks to #1142. @hairmare did you have a chance to test this?

I roughly tested the micro-site player and that the default stream still works on a fresh install. I wasn't able to test the "Playout History" changes since the branch I'm currently on has way to much unrelated changes to that controller/view.

**Comment by hairmare** [comment]
I figured out what the resizing code is supposed to do. It takes care of the popup from the listen button underneath the on air sign.

With this change the window looks a bit to large:

image

With the code working it should get resized to something like this:

image

I'm pretty sure that there are better ways to acheive this nowadays, but I haven't touched anything related to popups in modern HTML/JS recently so I'm not sure what the optimal solution would be here.

**Comment by zklosko** [comment]
Interesting... I'm also using Firefox and I haven't had a problem with the resizing.

**Comment by zklosko** [comment]
I really screwed my branch up. I'm going to start from scratch and make another PR when I'm ready. 😢

**Pull request closed without merging**